### PR TITLE
Minimize container dependency redistribution

### PR DESCRIPTION
**Pull request by rainersigwald**
Minimizing some of the transitive references was enabled by a slight refactoring that allows avoiding Cli.Utils for the net472 task.

Then avoid putting references already available in the SDK in the `net9.0` task assembly, and likewise trim STJ out of the `net472` one.

### CODE REVIEW

File: src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs [hunk 1]
```diff
@@ -51,7 +53,15 @@ private static string FindFullPathFromPath(string command)
...
-            string fullPath = Path.Combine(directory, command + FileNameSuffixes.CurrentPlatform.Exe);
+            string exeSuffix =
+#if NET
+                FileNameSuffixes.CurrentPlatform.Exe;
+#else
+                // Avoid dependencies from Microsoft.DotNet.Cli.Utils reference for VS/net472
+                RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : string.Empty;
+#endif
+
+            string fullPath = Path.Combine(directory, command + exeSuffix);
```
[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs:64]: I tend to agree but wanted to minimize the change. Would happily do this if SDK maintainers agree.

**Comment by rainersigwald** [comment]
This is failing on `PackageContentTest()`, which is true enough since I removed a bunch of stuff from the package. @baronfel, you mentioned dropping the creation of the package entirely. Should we do that here or should I fix up the test?

(@MichalPavlik's opinion also welcome)

File: src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (baronfel) [reply src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs:64]: Make it so!

**Review by baronfel — APPROVED** [review]

**Comment by baronfel** [comment]
Looks like there still one failing test that's checking the PackageReferences of the generated package - it thinks that STJ is still a dependency. This confuses me because that PackageReference is marked `ExcludeAssets="runtime"`, and I thought that metadata made the package invisible to NuGet's nuspec-generation.

**Comment by rainersigwald** [comment]
To make it explicit, I took the easy way out and just changed the expected-package-content test for now, but I do think #41703 is a good idea.

**Comment by rainersigwald** [comment]
> Looks like there still one failing test that's checking the PackageReferences of the generated package - it thinks that STJ is still a dependency. This confuses me because that PackageReference is marked `ExcludeAssets="runtime"`, and I thought that metadata made the package invisible to NuGet's nuspec-generation.

It is, however the test is actually just loading the XML of `src\Containers\Microsoft.NET.Build.Containers\Microsoft.NET.Build.Containers.csproj`! I was sorely tempted to fix this but resisted (since it should be deleted with the package).

**Comment by baronfel** [comment]
GREEN! Merging before someone breathes on this wrong.

**Pull request merged**